### ISSUE TITLE
PHP start fails: awk not found

### ISSUE THREAD

**Issue by alippert** [kyrofa/nextcloud-snap#11]
Distributor ID:	Ubuntu
Description:	Ubuntu 16.04 LTS
Release:	16.04
Codename:	xenial

<pre>sudo apt-get update && sudo apt-get upgrade -y
0 upgraded, 0 newly installed, 0 to remove and 0 not upgraded.</pre>

<pre>snap list
Name         Version               Rev  Developer  Notes
nextcloud    9.0.50snap2           5    canonical  -
ubuntu-core  16.04+20160531.11-56  122  canonical  -</pre>

<pre>Jun 22 14:50:09 ubuntu ubuntu-core-launcher[9192]: Waiting for mysql...
Jun 22 14:50:09 ubuntu ubuntu-core-launcher[9192]: Obtaining nextcloud mysql credentials...
Jun 22 14:50:09 ubuntu ubuntu-core-launcher[9192]: /snap/nextcloud/5/bin/start-php-fpm: 30: /snap/nextcloud/5/bin/start-php-fpm: awk: not found
Jun 22 14:50:09 ubuntu ubuntu-core-launcher[9192]: /snap/nextcloud/5/bin/start-php-fpm: 31: /snap/nextcloud/5/bin/start-php-fpm: arithmetic expression: expecting primary: "/50"
Jun 22 14:50:09 ubuntu systemd[1]: snap.nextcloud.php-fpm.service: Main process exited, code=exited, status=2/INVALIDARGUMENT
Jun 22 14:50:09 ubuntu systemd[1]: snap.nextcloud.php-fpm.service: Unit entered failed state.
Jun 22 14:50:09 ubuntu systemd[1]: snap.nextcloud.php-fpm.service: Failed with result 'exit-code'.
Jun 22 14:50:09 ubuntu systemd[1]: snap.nextcloud.php-fpm.service: Service hold-off time over, scheduling restart.
Jun 22 14:50:09 ubuntu systemd[1]: Stopped Service for snap application nextcloud.php-fpm.
Jun 22 14:50:09 ubuntu systemd[1]: snap.nextcloud.php-fpm.service: Start request repeated too quickly.
Jun 22 14:50:09 ubuntu systemd[1]: Failed to start Service for snap application nextcloud.php-fpm.
</pre>

TO DUPLICATE:

Simply install gawk, which will replace mawk in /etc/alternatives.

**Comment by kyrofa** [comment]
Thanks @alippert, this was just reported to me via email-- working on it now.

**Comment by alippert** [comment]
awk is mapped via etc.alternatives to gawk and is functioning at the command line:

ubuntu@ip-10-10-0-89:~$ awk '/MemTotal/ {printf "%d", $2/1024}' /proc/meminfo
3950

**Comment by kyrofa** [comment]
@alippert can you run this test for me?

```bash
$ sudo snap install hello-world
$ hello-world.sh
```

That will drop you into a confined shell. Now please run:

```bash
$ which awk
$ awk '/MemTotal/ {printf "%d", $2/1024}' /proc/meminfo
```

**Comment by alippert** [comment]
Interesting:

`ubuntu@ip-10-10-0-89:~$ hello-world.sh
Launching a shell inside the default app confinement. Navigate to your
app-specific directories with:

  $ cd $SNAP
  $ cd $SNAP_DATA
  $ cd $SNAP_USER_DATA

bash-4.3$ which awk
bash-4.3$ awk '/MemTotal/ {printf "%d", $2/1024}' /proc/meminfo
bash: awk: command not found
bash-4.3$ 
`

**Comment by kyrofa** [comment]
Interesting indeed, thank you. I thought awk was included in the core. Investigating now. To clarify, you're running vanilla Ubuntu 16.04? Not a derivative?

**Comment by alippert** [comment]
Correct.

> ubuntu@ip-10-10-0-89:~$ which awk
/usr/bin/awk

But apparently not in core.

**Comment by kyrofa** [comment]
Okay, one more test (sorry): what is the output of

```bash
$ snap list
```

**Comment by alippert** [comment]
<pre>Name         Version               Rev  Developer  Notes
hello-world  6.1                   26   canonical  -
nextcloud    9.0.50snap2           5    canonical  -
ubuntu-core  16.04+20160531.11-56  122  canonical  -
</pre>

FYI: https://bugs.launchpad.net/ubuntu/+source/snapd/+bug/1580018

**Comment by kyrofa** [comment]
Good find @alippert, I'll just bundle it for now.

**Comment by alippert** [comment]
Thanks

**Comment by kyrofa** [comment]
@alippert can you verify that this was resolved in version 9.0.50snap3?

**Comment by alippert** [comment]
Snap 3 looks good. Nice work - and _fast_ too!
